keps/sig-release/2572-release-cadence/README.md

```diff
@@ -0,0 +1,440 @@
+for 2022, leveraging the historical *4-releases-per-year cadence*. Generally, 
+code freeze remains in effect until the last week of the release, so 
+development for the next release generally starts prior to the official release 
+team kickoff. A minimum of 1 week is needed between releases to fully form the 
+release team and to facilitate on-boarding of shadows. The fourth release of 
+the year has traditionally been compressed and limited in scope, overlapping 
+with end of year holidays and vacation for many contributors. Additionally, 
+KubeCon normally occurs during at least one release, eliminating at *least* one week of 
+working time.
+
+*Kubernetes Release Sechedule 2021 (Existing 4 Release Cadence)*
+
+| Year Week Number | Release Number | Release Week | Note |
+| -------- | -------- | -------- | -------- |
+| 2     | 1     | 1 (January 11) | |
+| 14 | 1 | 13 (April 8) | |
+| 16 | 2 | 1 (April 19) | |
+| 27 | 2 | 11 (July 06) | One week break for KubeCon EU | 
+| 29 | 3 | 1 (July 20) | |
+| 40 | 3 | 11 (October 5) | | 
+| 42 | 4 | 1 (October 18) | |
+| 52 | 4 | 10 (December 28) | End of Year Holidays |
+
+*Kubernetes Release Sechedule 2022 (Existing 4 Release Cadence)*
+
+| Year Week Number | Release Number | Release Week | Note |
+| -------- | -------- | -------- | -------- |
+| 1  | 1 | 1 (January 3) | |
+| 12 | 1 | 12 (March 15) | |
+| 14 | 2 | 1 (March 28) | Probable KubeCon EU Break|
+| 26 | 2 | 12 (June 28) | |
+| 28 | 3 | 1 (July 11) | | 
+| 40 | 3 | 12 (October 4) | | 
+| 42 | 4 | 1 (October 17) | Probably KubeCon NA Break | 
+| 52 | 4 | 10 (Dec 28) | | 
+
+With the proposed change in cadence, the notional schedules for the remainder of
+2021 and 2022 are shown below:
+
+*Kubernetes Release Schedule 2021 (Proposed 3 Release Cadence)*
+
+| Year Week Number | Release Number | Release Week | Note |
+| -------- | -------- | -------- | -------- |
+| 2  | 1 | 1 (January 11) | |
+| 14 | 1 | 13 (April 8) | |
+| 17 | 2 | 1 (April 26) | |
+| 32 | 2 | 15 (August 02) | KubeCon EU Break (May 4-7) |
+| 35 | 3 | 1 (August 30) | | 
+| 50 | 3 | 15 (December 14) | Kubecon Break (Oct 12-15) | 
+
+*Kubernetes Release Sechedule 2022 (Proposed 3 Release Cadence)*
+
+| Year Week Number | Release Number | Release Week | Note |
+| -------- | -------- | -------- | -------- |
+| 1  | 1 | 1 (January 3) | |
+| 15 | 1 | 15 (April 12) | | 
+| 18 | 2 | 1 (May 2) | Probably KubeCon EU |
+| 33 | 2 | 15 (August 15) | |
+| 36 | 3 | 1 (September 6 | Probably KubeCon NA | 
+| 51 | 3 | 15 (December 20) | 
+
+
+This KEP will be in the `alpha` stage for the Kubernetes 1.22 Release. During 
+this time, SIG-Release will focus on communication of the cadence change through
+all available mechanisms. The KEP will promote to the `beta` stage for the 
+Kubernetes 1.23 Release, which will be the final release of 2021. After 
+the 1.23, 1.24, and  1.25 Releases, will will collect feedback and incorporate 
```

Comment:
@jeremyrickard Given that we already have feedback that this cadence may not work for people and given how hectic and disruptive mid-year changes are, wondering if it is better to try to do an EOY survey and have this finalized before moving into 2022? The current setup makes it feel v difficult for people to be able actually revert this if it goes wrong until 2023 (since 2/3 of the year would be over by the time the decision is made and quarters dont fit into thirds very well).

Reply:
I agree that making the transition mid 2022 would not be great. The best approach to this and handling a revert is probably to approach it in a manner where we evaluate during the releases listed and then during the 1.26 release, we would either graduate this to stable OR we would take another action (revert to 4 releases per year, entertain 2 releases per year, entertain 10 releases per year), but such a cadence change would not start until the first release of 2023, to allow adequate communication and avoid any midyear confusion / pain. WDYT? I think running the surveys for several releases with accompanying communication will allow us to gather appropriate data and compare to cycles prior to 1.19?

Reply:
Added a couple of sentences to address and clarify this @kikisdeliveryservice

Reply:
Thanks for the update @jeremyrickard I think it's more honest about the ramifications of the current state (this isn't going to revert back midyear as it's very disruptive). One thing that troubles me is that if there are serious objections/hurdles that show this just isn't working out in alpha/early beta, there isn't a way to revert for the beginning of 2022. I'm not sure if anyone else shares this concern, but wanted to raise it.

Reply:
So:  

Evaluation mid-2022 (1.25), decision on permanence (until we change it again, anyway) to take effect Jan 2023 (1.27)?

Reply:
The alternative is to eval late 2021.  Question is whether we'll feel ready to evaluate it then.

Reply:
We could try to eval late 2021, if we have sufficient data. If we want to provide ample notification time (6 months or more), how do we handle the change back? We would probably need to hold the first release to 15 weeks for predictability, then shift to 10-11 week releases?

Reply:
Do we have a few tentative questions that we plan to ask in that survey (whenever it is decided to be sent out)? For example, one clear question is "Do you think the community should go forward with 3 releases/year? Yes/No/Maybe". We may have more on the lines of reviewer bandwidth.

What else can be the metrics which we gauge in the survey so that we can make an effective decision about the future of release cadence (continue or revert)?

Reply:
> o you think the community should go forward with 3 releases/year? Yes/No/Maybe"

That is not a good question for a survey, because you are forcing to take a decision on something predetermined.

For this to work we should take into account the profile of the surveyed person, i.e. making up the numbers, if the Kubernetes market is 30% cloud providers, 30% software vendors, 30% in-house and 10% weekend hackers , we can't decide the release cadence on the latest ...

We also should get some information to be able to ponder the answer correctly: 
do you use your cluster in production? how many nodes?
A person with 2 nodes and nothing in production will not have the same problems that someone with 50 nodes with critical services in their company.

What is at stake here is the release model, Tim Hockin has a good comment exploring different alternative models, and we should get feedback on this, we are assuming that we want a fixed release model, but is on us to innovate/improve and think out of the box, what problem do we want to solve? how can we do it?

Coming back to the survey, I think that the question should be something like

> Kubernetes projects need your feedback to choose a release model, which one do you prefer and why?
> 1. 3 timeboxed releases per year and and EOL cycle of 1 year
> 2. 4 timeboxed releases per year
> 3. tick - tock release
> 4.  ....
> 5. blank so user can say why those models doesn't work and which one he suggests

Reply:
let's please not litigate the contents of a survey we would send out later. If needed we can open another issue or a google doc and design the survey there.

Reply:
Created an issue in https://github.com/kubernetes/sig-release/issues/1526, feel free to comment there if I missed anything from the conversation. 

Resolving this discussion for now, let's focus on the KEP itself.